File: FirstScreen/Sources/FirstScreenFactory.m
//
//  FirstScreenFactory.m
//  FirstScreenImpl
//
//  Created by Maxim on 16.12.2024.
//

@import FirstScreenInterface;
#import <FirstScreenImpl/FirstScreenImpl-Swift.h>

@implementation FirstScreenFactory
+ (UIViewController * _Nonnull)makeFirstScreenWith:(id<FirstScreenAppearance> _Nullable)appearance {
    return [FirstScreenFactoryImpl makeFirstScreen:appearance];
}

@end

File: FirstScreen/Sources/FirstScreenFactory.swift
//
//  FirstScreenFactoryImpl.swift
//  FirstScreenImpl
//
//  Created by Vasiliy Fedotov on 17.11.2024.
//

import UIKit
import FirstScreenInterface
import SecondScreenInterface

@objc public
final class FirstScreenFactoryImpl: NSObject {

    @objc public
    static func makeFirstScreen(_ appearance: FirstScreenAppearance?) -> UIViewController {
        let appearance = appearance ?? FirstScreenAppearanceFactory().makeAppearance(forPrefferedId: 1)
        let screen2Factory = SecondScreenFactory()

        let router = FirstScreenRouter(secondScreenFactory: screen2Factory)
        let presenter = FirstScreenPresenterImpl(router: router)

        let vc = FirstScreenViewController(appearance: appearance, presenter: presenter)

        presenter.view = vc
        router.fromVC = vc

        return vc
    }
}
